### PR TITLE
storage: Remove ReadOnlyLogTX interface

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
The only used method of `ReadOnlyLogTX` was `GetActiveLogIDs`. This change moves this
method directly to  the `LogStorage` interface without affecting transaction semantics.

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: log/operation_manager.go [hunk 1]
```diff
@@ -158,18 +158,10 @@ func NewOperationManager(info OperationInfo, logOperation Operation) *OperationM
 
 // getActiveLogIDs returns IDs of logs eligible for sequencing.
 func (o *OperationManager) getActiveLogIDs(ctx context.Context) ([]int64, error) {
-	tx, err := o.info.Registry.LogStorage.Snapshot(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create transaction: %v", err)
-	}
-	defer tx.Close()
-	logIDs, err := tx.GetActiveLogIDs(ctx)
+	logIDs, err := o.info.Registry.LogStorage.GetActiveLogIDs(ctx)
```
Comment (pphaneuf) [inline log/operation_manager.go:161]: This is so tantalisingly close to being just `return o.info.Registry.LogStorage.GetActiveLogIDs(ctx)`... 😃

Comment (pavelkalinnikov) [reply log/operation_manager.go]: Inlined this invocation.

File: storage/log_storage.go [hunk 2]
```diff
@@ -102,9 +87,7 @@ type LogTreeTX interface {
 // ReadOnlyLogStorage represents a narrowed read-only view into a LogStorage.
 type ReadOnlyLogStorage interface {
 	DatabaseChecker
-
-	// Snapshot starts a read-only transaction not tied to any particular tree.
-	Snapshot(ctx context.Context) (ReadOnlyLogTX, error)
+	LogMetadata
```
Comment (pphaneuf) [inline storage/log_storage.go:90]: I don't think there's any mention of this interface, so what would you think of inlining the method here, and deleting `LogMetadata` entirely?

**Review by pphaneuf — APPROVED** [review]

File: storage/log_storage.go (hunk 2 above)
Comment (pavelkalinnikov) [reply storage/log_storage.go:90]: SGTM. I'll do it in a follow-up PR if you don't mind?

Comment (pphaneuf) [reply storage/log_storage.go:90]: Absolutely, no problem.

Comment (pavelkalinnikov) [reply storage/log_storage.go:90]: https://github.com/google/trillian/pull/2550

**Pull request merged**